### PR TITLE
Remove FEniCS as requirement

### LINKED ISSUE #103: Installing fenics adapter may break FEniCS installation
See https://fenicsproject.discourse.group/t/installing-python-package-with-fenics-dependency-breaks-fenics-installation/4476.

To me this looks like an upstream issue upstream. The fenics adapter depends on fenics. Therefore the following dependencies in `setup.py` look correct:

https://github.com/precice/fenics-adapter/blob/fcd3f0c50b030b35cbea54b278d7c015a67447cd/setup.py#L28

However, if we assume a user has already fixed the FEniCS installation by manually uninstalling `fenics-ufl`, the fenics adapter will reinstall `fenics-ufl` due to the `fenics` dependency and (again) break the FEniCS installation. The user has to uninstall `fenics-ufl` (again) after installing the fenics adapter.

A possible solution that I have in mind is to declare FEniCS as a dependency of the adapter, but avoid reinstalling `fenics-ufl`. Is this possible?

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
Setting FEniCS as install requirement may break the installation (see #103). We therefore check whether FEniCS is installed and abort, if this is not the case.

Closes #103.

@fsimonis @IshaanDesai What do you think about this approach?

### CODE REVIEW

**Review by IshaanDesai — APPROVED** [review]
For other adapters (OpenFOAM, CalculiX, etc.) the solver installation is something that the user takes care of and is not a part of the adapter installation process, hence this change goes in the right direction. :+1:

**Review by fsimonis — APPROVED** [review]
As the apt package does not install the full stack, this is the only sane way forward.
I would extend the error message a bit though.

**Comment by BenjaminRodenberg** [comment]
See #118. Systemtest is currently not working due to an issue in the systemtests repository.

File: setup.py [hunk 1]
```diff
@@ -7,6 +7,15 @@
 if sys.version_info[0] < 3:
     raise Exception("fenicsprecice only supports Python3. Did you run $python setup.py <option>.? Try running $python3 setup.py <option>.")
 
+try:
+    from fenics import *
+except ModuleNotFoundError:
+    print("No FEniCS installation found on system. Please install FEniCS and check whether it is found correctly.\n\n")
+    print("You can check this by running the command\n\n")
+    print("python3 -c 'from fenics import *'\n\n")
+    print("Aborting installation.")
```
[the remark this answers is not in the document]
Comment (BenjaminRodenberg) [reply setup.py:18]: @fsimonis I removed the statement `"Note that "apt install fencis" will N O T install the full required software stack!"` in https://github.com/precice/fenics-adapter/pull/130/commits/3024e32708cc0c863f6699fbe67b1332a7ac6082. I should probably already have asked here, but can you elaborate on this? What do you mean? What is missing? We can reintroduce this statement or a similar one. Currently it's [not clear](https://github.com/precice/fenics-adapter/pull/130#issuecomment-825778811) and I also don't fully understand it.

**Pull request merged**